oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/register/ServiceInventory.java

```diff
@@ -187,7 +194,7 @@ public ServiceInventory getClone() {
         if (serviceInventory.getLastUpdateTime() >= this.getLastUpdateTime()) {
             this.nodeType = serviceInventory.getNodeType();
             setProp(serviceInventory.getProp());
-            if (Const.NONE != serviceInventory.getMappingServiceId()) {
+            if (serviceInventory.isForceResetServiceMapping() || Const.NONE != serviceInventory.getMappingServiceId()) {
```

Comment:
`#forceResetServiceMapping` should be combined, otherwise, if some combine merge happens in streaming, rather than memory-db merge, this field lost, as a result, the database would update `mappingServiceId` anymore.